lib/dal/src/workspace_snapshot/split_snapshot/prop.rs

```diff
@@ -0,0 +1,63 @@
+use async_trait::async_trait;
+use si_id::{
+    AttributePrototypeId,
+    PropId,
+};
+
+use crate::{
+    DalContext,
+    EdgeWeightKindDiscriminants,
+    prop::{
+        PropError,
+        PropResult,
+    },
+    workspace_snapshot::{
+        split_snapshot::SplitSnapshot,
+        traits::{
+            attribute_prototype::AttributePrototypeExt as _,
+            attribute_prototype_argument::AttributePrototypeArgumentExt as _,
+            func::FuncExt as _,
+            prop::PropExt,
+        },
+    },
+};
+
+#[async_trait]
+impl PropExt for SplitSnapshot {
+    async fn prop_default_value(
+        &self,
+        ctx: &DalContext,
+        prop_id: PropId,
+    ) -> PropResult<Option<serde_json::Value>> {
+        let prototype_id = self.prop_prototype_id(prop_id).await?;
+        let func_id = self.attribute_prototype_func_id(prototype_id).await?;
+        if self.func_is_dynamic(func_id).await? {
+            return Ok(None);
+        }
+
+        match self
+            .attribute_prototype_arguments(prototype_id)
+            .await?
+            .first()
+        {
+            Some(&apa_id) => self
+                .attribute_prototype_argument_static_value(ctx, apa_id)
+                .await
+                .map_err(Into::into),
```

Comment:
`?` with surrounding `Ok()` will likely end up more compact here (at the least, the formatter is willing to turn that into a single line if it's small enough).